include/jemalloc/internal/rtree_inlines.h

```diff
@@ -161,11 +163,17 @@ rtree_elm_acquire(tsdn_t *tsdn, rtree_t *rtree, rtree_ctx_t *rtree_ctx,
 
 	spin_t spinner = SPIN_INITIALIZER;
 	while (true) {
-		extent_t *extent = rtree_elm_read(elm, false);
 		/* The least significant bit serves as a lock. */
-		void *s = (void *)((uintptr_t)extent | (uintptr_t)0x1);
-		if (!atomic_cas_p(&elm->pun, (void *)extent, s)) {
-			break;
+		void *extent_and_lock = atomic_load_p(&elm->child_or_extent,
+		    ATOMIC_RELAXED);
+		if (likely(((uintptr_t)extent_and_lock & 0x1) == 0)) {
```

Comment:
For my own edification, why do we need to cast 0x1 to uintptr_t? Is it because Clang warns on using signed and unsigned values together? If so, wouldn't 0x1u be sufficient?